### DISCUSSION TITLE
Text2DBundle ComputedVisibility not working?

### DISCUSSION THREAD

**Discussion by Liam19** [bevyengine/bevy#9745] [Q&A] [answered]
Im trying to add "Label" entities to my main game entities using a Text2DBundle. 
Currently i have 100s of labels and it's causing a huge slowdown even when they are off screen. I made a system that checks each label's ComputedVisibilty using the is_visible_in_view() function, and counts up the number of visible labels. 
For some reason, the count is always the same even when moving the camera? It seems like the computed visibility is not being changed at all.

**Comment by ickshonpe** [comment, marked as answer]
I've been meaning to look at this but haven't got around to it.

There is a second issue as well: Text2d rendering isn't very optimised and it just extracts an individual sprite for each glyph. So with hundreds of labels you could be extracting tens of thousands of sprites.

As a temporary fix you could set up your own visibility system, `Text2dBundle` has a `TextLayoutInfo` component with a size field that contains the size of the text. You can use that to create a bounding box which you can test to if it's on screen and if not set it non-visible yourself.

**Reply by Liam19** [reply]
Do you know of any PRs that address this issue? Im willing to wait for an 'official' solution if there is one coming.
